payment_ippay_ach/models/payment.py

```diff
@@ -12,22 +13,61 @@
 
 
 class PaymentAcquirer(models.Model):
+    """PaymentAcquirer."""
+
     _inherit = "payment.acquirer"
 
-    provider = fields.Selection(selection_add=[("ippay_ach", "IPPay ACH eCheck")])
+    provider = fields.Selection(selection_add=[(
+        "ippay_ach", "IPPay ACH eCheck")])
     api_url = fields.Char("Api URL", required_if_provider="ippay_ach")
     ippay_ach_terminal_id = fields.Char(
         "IPpay ACH TerminalID", required_if_provider="ippay_ach"
     )
 
+    @api.model
+    def ippay_ach_s2s_form_process(self, data):
+        """Create the token when user add the details from customer Portal."""
+        token_id = data.get('selected_token_id')
+        if not token_id:
+            values = {
+                'bank_acc_number': data.get('bank_acc_number'),
+                'aba': data.get('aba'),
+                'ch_holder_name': data.get('ch_holder_name'),
+                'acquirer_id': int(data.get('acquirer_id')),
+                'partner_id': int(data.get('partner_id'))
+            }
+            payment_method = self.env['payment.token'].sudo().create(values)
```

Comment:
Please incorporate here the logic to "Only create a new Token if it doesn't alerady exist" - see payment_ippay for reference.